src/uu/cat/src/cat.rs

```diff
@@ -804,21 +790,36 @@ mod tests {
     #[test]
     fn test_incrementing_string() {
         let mut incrementing_string = super::LineNumber::new();
-        assert_eq!(b"     1\t", incrementing_string.buf.as_slice());
+        assert_eq!(
+            b"     1\t",
+            &incrementing_string.buf[incrementing_string.print_start..]
```

Comment:
Now that I see those not-so-nice changes here, I wonder if I should instead create some struct to encapsulate the buffer, start/end position (basically, similar to what `LineNumber` does here, but move to `uucore`)

Reply:
I found it difficult to merge code, as the logic in `cat` and `seq` is quite custom and different (different padding logic, etc...). We could try to merge if we have more callers, otherwise I think it's good as is.

One thing I changed though is to pass `start` as a `&mut`, makes the calling API a little bit nicer IMHO.